src/mainboard/sunxi/nezha/bt0/src/main.rs

```diff
@@ -42,26 +42,10 @@ const STACK_SIZE: usize = 1 * 1024; // 1KiB
 #[link_section = ".bss.uninit"]
 static mut BT0_STACK: [u8; STACK_SIZE] = [0; STACK_SIZE];
 
-/// Jump over head data to executable code.
-///
-/// # Safety
-///
-/// Naked function.
-#[naked]
-#[link_section = ".head.text"]
-#[export_name = "_head_jump"]
-pub unsafe extern "C" fn head_jump() {
-    asm!(
-        ".option push",
-        ".option rvc",
-        "c.j    0x68", // 0x60: eGON.BT0 header; 0x08: FlashHead
-        ".option pop",
-        // sym _start,
-        options(noreturn)
-    )
-}
-
-// todo: option(noreturn) generates an extra `unimp` insn
+// Use assembly file for a 4 byte jump instruction to _start.
+// The reason is that rust adds an extra `unimp` insn after the jump.
+// This messes up the location of the eGON header
```

Comment:
I cannot confirm this.
The current code works and has been working for years now.
How did you experience the issue described here?

Reply:
The current instruction is compressed, on that note, i.e. 2 bytes, and we need 4, so the other 2 bytes are necessary.

Reply:
So the current code jumps a bit too far. "// 0x60: eGON.BT0 header; 0x08: FlashHead" Is not correct. Flashhead is not part of the image so the jump should be to 0x60, not 0x68. I don't think disabling interrupts (might be reset default) and enabling the Thead specific extensions are crucial, which is why things worked. Using a 4 byte jump, just like u-boot (4 first bytes are exactly the same) seemed convenient.

I tried tried inline assembly jump to a symbol, rather than a raw number, but then the rust compiler pads the jump too much with another 4 bytes of nothing. This is what my comment meant.

Reply:
Oooh... gotcha; so can't we just change the 0x68 to 0x60 then?
I'd prefer the inlining so that it's all in one place, but of course it should also be correct.

Reply:
```
Disassembly of section .head:

0000000000020000 <_head_jump>:
   20000:       a0a5                    j       20068 <_start+0x8>
        ...

0000000000020004 <_ZN17oreboot_nezha_bt09EGON_HEAD17h20872eadf5e41c7cE>:
   20004:       4765                    li      a4,25
   20006:       422e4e4f                .4byte  0x422e4e4f
   2000a:       3054                    .2byte  0x3054
   2000c:       6c39                    lui     s8,0xe
   2000e:       5f0a                    lw      t5,160(sp)
        ...

Disassembly of section .text:

0000000000020060 <_start>:
   20060:       30401073                csrw    mie,zero
   20064:       00600337                lui     t1,0x600
   20068:       7c032073                csrs    0x7c0,t1
   2006c:       000303b7                lui     t2,0x30
   20070:       23cd                    addiw   t2,t2,19 # 30013 <ebss+0xbf68>
   20072:       7c239073                csrw    0x7c2,t2
```

this is what currently happens, yea

Reply:
and what you're really saying is that MainStageHead was never effectively used - and it is probably not needed?

Reply:
https://github.com/u-boot/u-boot/blob/fe2ce09a0753634543c32cafe85eb87a625f76ca/include/sunxi_image.h#L35

looking at this, it appears that most of the data (all 0 in our current code) isn't used anyway... funny

Reply:
> Oooh... gotcha; so can't we just change the 0x68 to 0x60 then? I'd prefer the inlining so that it's all in one place, but of course it should also be correct.

As the file indeed contains just 1 instruction, I moved it to using strings in the main.rs file.